docs/source/use-cases.rst

```diff
@@ -0,0 +1,375 @@
+NumPy), dataframes (similar to Pandas), and lists (similar to PyToolz or
+PySpark.)
+
+On disk arrays
+~~~~~~~~~~~~~~
+
+Scientists studying the earth have 10GB to 100GB of regularly gridded weather
+data on their laptop's hard drive stored as many individual HDF5 or NetCDF
+files.  They use :doc:`dask.array<array>` to treat this stack of HDF5 or NetCDF
+files as a single NumPy_ array (or a collection of NumPy arrays with the
+XArray_ project).  They slice, perform reductions, perform seasonal averaging
+etc. all with straight Numpy syntax.  These computations take a few minutes
+(reading 100GB from disk is somewhat slow) but previously infeasible
+computations become convenient from the comfort of a personal laptop.
+
+It's not so much parallel computing that is valuable here but rather the
+ability to comfortably compute on larger-than-memory data without special
+hardware.
+
+.. code-block:: python
+
+    import h5py
+    dataset = h5py.File('myfile.hdf5')['/x']
+
+    import dask.array as da
+    x = da.from_array(dataset, chunks=dataset.chunks)
+
+    y = x[::10] - x.mean(axis=0)
+    y.compute()
+
+.. _NumPy: http://www.numpy.org/
+.. _XArray: http://xarray.pydata.org/en/stable/
+
+Directory of CSV or tabular HDF files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Analysts studying time series data have a large directory of CSV, HDF, or
+otherwise formatted tabular files.  They usually use Pandas_ for this kind of
+data but either the volume is too large or dealing with a large number of files
+is confusing.  They use :doc:`dask.dataframe<dataframe>` to logically wrap all
+of these different files into one logical dataframe that is built on demand to
+save space.  Most of their Pandas workflow is the same (Dask.dataframe is a
+subset of Pandas) so they switch from Pandas to Dask.dataframe and back easily
+without significantly changing their code.
+
+.. code-block:: python
+
+   import dask.dataframe as dd
+
+   df = dd.read_csv('data/2016-*.*.csv', parse_dates=['timestamp'])
+   df.groupby(df.timestamp.dt.hour).value.mean().compute()
+
+.. _Pandas: http://pandas.pydata.org/
+
+Directory of CSV files on HDFS
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The same analyst as above uses dask.dataframe with the dask.distributed_ scheduler
+to analyze terabytes of data on their institution's Hadoop cluster straight
+from Python.  This uses the HDFS3_ Python library for HDFS management
+
+This solution is particularly attractive because it stays within the Python
+ecosystem, and uses the speed and algorithm set of Pandas_, a tool with which
+the analyst is already very comfortable.
+
+.. code-block:: python
+
+   from dask.distributed import Executor
+   e = Executor('cluster-address:8786')
+
+   import dask.dataframe as dd
+   df = dd.read_csv('hdfs://data/2016-*.*.csv', parse_dates=['timestamp'])
+   df.groupby(df.timestamp.dt.hour).value.mean().compute()
+
+.. _HDFS3: http://hdfs3.readthedocs.io/en/latest/
+
+
+Directories of custom tabular files
```

Comment:
Perhaps use "custom delimited files" or just "custom format files" instead. Pandas has a built in tabular file reader `read_table` which is pretty flexible in regards to delimiter, so this use case may not be as apparent. It also occurs to me that we might as well wrap `read_table`, since it's almost exactly the same as `read_csv`.